managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java

```diff
@@ -233,84 +224,76 @@ public void asyncUpdateCursorInfo(final String ledgerName, final String cursorNa
                 info.getCursorsLedgerId(), info.getMarkDeleteLedgerId(), info.getMarkDeleteEntryId());
 
         String path = prefix + ledgerName + "/" + cursorName;
-        byte[] content = info.toString().getBytes(Encoding);
+        byte[] content = info.toByteArray();
```

Comment:
Nope, this is using the binary format, and it's intentional, I mentioned it in the description of the PR.
Text format is REALLY verbose, and its size grows A LOT for each entry.
Here's what an entry with only 71 messages looks like:
```
cursorsLedgerId: -1 markDeleteLedgerId: 459 markDeleteEntryId: 59 individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 60 } upperEndpoint { ledgerId: 459 entryId: 1845 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1846 } upperEndpoint { ledgerId: 459 entryId: 1848 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1849 } upperEndpoint { ledgerId: 459 entryId: 1851 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1852 } upperEndpoint { ledgerId: 459 entryId: 1854 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1855 } upperEndpoint { ledgerId: 459 entryId: 1857 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1858 } upperEndpoint { ledgerId: 459 entryId: 1860 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1861 } upperEndpoint { ledgerId: 459 entryId: 1863 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1864 } upperEndpoint { ledgerId: 459 entryId: 1866 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1867 } upperEndpoint { ledgerId: 459 entryId: 1869 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1870 } upperEndpoint { ledgerId: 459 entryId: 1872 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1873 } upperEndpoint { ledgerId: 459 entryId: 1875 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1876 } upperEndpoint { ledgerId: 459 entryId: 1878 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1879 } upperEndpoint { ledgerId: 459 entryId: 1881 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1882 } upperEndpoint { ledgerId: 459 entryId: 1884 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1885 } upperEndpoint { ledgerId: 459 entryId: 1887 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1888 } upperEndpoint { ledgerId: 459 entryId: 1890 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1891 } upperEndpoint { ledgerId: 459 entryId: 1893 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1894 } upperEndpoint { ledgerId: 459 entryId: 1896 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1897 } upperEndpoint { ledgerId: 459 entryId: 1899 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1900 } upperEndpoint { ledgerId: 459 entryId: 1902 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1903 } upperEndpoint { ledgerId: 459 entryId: 1905 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1906 } upperEndpoint { ledgerId: 459 entryId: 1908 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1909 } upperEndpoint { ledgerId: 459 entryId: 1911 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1912 } upperEndpoint { ledgerId: 459 entryId: 1914 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1915 } upperEndpoint { ledgerId: 459 entryId: 1917 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1918 } upperEndpoint { ledgerId: 459 entryId: 1920 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1921 } upperEndpoint { ledgerId: 459 entryId: 1923 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1924 } upperEndpoint { ledgerId: 459 entryId: 1926 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1927 } upperEndpoint { ledgerId: 459 entryId: 1929 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1930 } upperEndpoint { ledgerId: 459 entryId: 1932 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1933 } upperEndpoint { ledgerId: 459 entryId: 1935 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1936 } upperEndpoint { ledgerId: 459 entryId: 1938 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1939 } upperEndpoint { ledgerId: 459 entryId: 1941 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1942 } upperEndpoint { ledgerId: 459 entryId: 1944 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1945 } upperEndpoint { ledgerId: 459 entryId: 1947 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1948 } upperEndpoint { ledgerId: 459 entryId: 1950 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1951 } upperEndpoint { ledgerId: 459 entryId: 1953 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1954 } upperEndpoint { ledgerId: 459 entryId: 1956 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1957 } upperEndpoint { ledgerId: 459 entryId: 1959 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1960 } upperEndpoint { ledgerId: 459 entryId: 1962 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1963 } upperEndpoint { ledgerId: 459 entryId: 1965 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1966 } upperEndpoint { ledgerId: 459 entryId: 1968 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1969 } upperEndpoint { ledgerId: 459 entryId: 1971 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1972 } upperEndpoint { ledgerId: 459 entryId: 1974 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1975 } upperEndpoint { ledgerId: 459 entryId: 1977 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1978 } upperEndpoint { ledgerId: 459 entryId: 1980 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1981 } upperEndpoint { ledgerId: 459 entryId: 1983 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1984 } upperEndpoint { ledgerId: 459 entryId: 1986 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1987 } upperEndpoint { ledgerId: 459 entryId: 1989 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1990 } upperEndpoint { ledgerId: 459 entryId: 1992 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1993 } upperEndpoint { ledgerId: 459 entryId: 1995 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1996 } upperEndpoint { ledgerId: 459 entryId: 1998 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 1999 } upperEndpoint { ledgerId: 459 entryId: 2001 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2002 } upperEndpoint { ledgerId: 459 entryId: 2004 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2005 } upperEndpoint { ledgerId: 459 entryId: 2007 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2008 } upperEndpoint { ledgerId: 459 entryId: 2010 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2011 } upperEndpoint { ledgerId: 459 entryId: 2013 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2014 } upperEndpoint { ledgerId: 459 entryId: 2016 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2017 } upperEndpoint { ledgerId: 459 entryId: 2019 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2020 } upperEndpoint { ledgerId: 459 entryId: 2022 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2023 } upperEndpoint { ledgerId: 459 entryId: 2025 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2026 } upperEndpoint { ledgerId: 459 entryId: 2028 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2029 } upperEndpoint { ledgerId: 459 entryId: 2031 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2032 } upperEndpoint { ledgerId: 459 entryId: 2034 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2035 } upperEndpoint { ledgerId: 459 entryId: 2037 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2038 } upperEndpoint { ledgerId: 459 entryId: 2040 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2041 } upperEndpoint { ledgerId: 459 entryId: 2043 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2044 } upperEndpoint { ledgerId: 459 entryId: 2046 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2047 } upperEndpoint { ledgerId: 459 entryId: 2048 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2050 } upperEndpoint { ledgerId: 459 entryId: 2051 } } individualDeletedMessages { lowerEndpoint { ledgerId: 459 entryId: 2053 } upperEndpoint { ledgerId: 459 entryId: 2054 } }
```
That's about 8k, and this is only going to get larger, larger Ids mean larger string size.

I know that introducing a change in format is a breaking change, but I changed usages of this to parse both formats.